src/main/java/com/theelixrlabs/healthcare/validation/Validator.java

```diff
@@ -0,0 +1,61 @@
+package com.theelixrlabs.healthcare.validation;
+
+import com.theelixrlabs.healthcare.constants.PatientConstants;
+import com.theelixrlabs.healthcare.dto.PatientDTO;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import org.springframework.context.MessageSource;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class Validator {
+
+    private final MessageSource messageSource;
+
+    public Validator(MessageSource messageSource) {
+        this.messageSource = messageSource;
+    }
+
+    /**
+     * Validates and converts a given string representation of UUID into a UUID object.
+     * Throws a CustomException with a specific error message if the string is not a valid UUID format.
+     *
+     * @param id    The string representation of UUID to validate and convert.
+     * @param errorMessage    The error message to use in the CustomException if validation fails.
+     * @return The UUID object parsed from the input string.
+     * @throws CustomException    If the input string is not a valid UUID format.
+     */
+    public UUID validateUUID(String id, String errorMessage) throws CustomException {
```

Comment:
this is not just validation the id it is converting String id to UUID type and returning it so please rename accordingly and update the description

Reply:
Updated.